### PR TITLE
Add an error message indicating SPHInX is not installed

### PR DESCRIPTION
**Pull request by samwaseda**
Based on [this discussion](https://github.com/pyiron/pyiron/pull/1479) with @liamhuber, I added an error message in SPHInX (Liam's inner voice: "we were talking about GPAW!!").

I put it in the log parser, because that's where the error is usually shown first, but to be honest I find is a bit clumsy, because it's more the symptom and not the error itself. If there's someone who has a better idea, let me know.

### CODE REVIEW

**Comment by pmrv** [comment]
I solved a similar problem with [atomsk](https://github.com/pyiron/pyiron_atomistics/blob/f6007ccb38172a4cd766ff2a7e0614d9cefa6f4a/pyiron_atomistics/atomistics/structure/factories/atomsk.py#L27) using `shutil.which`.  I guess if you do that check in the sphinx top level module or in `SphinxBase.__init__` you can fail earlier.

**Comment by samwaseda** [comment]
> I solved a similar problem with [atomsk](https://github.com/pyiron/pyiron_atomistics/blob/f6007ccb38172a4cd766ff2a7e0614d9cefa6f4a/pyiron_atomistics/atomistics/structure/factories/atomsk.py#L27) using `shutil.which`. I guess if you do that check in the sphinx top level module or in `SphinxBase.__init__` you can fail earlier.

!? God Marvin you totally knocked my eyes off!!!

**Comment by samwaseda** [comment]
But where's `_ATOMSK_EXISTS` actually used?

**Comment by pmrv** [comment]
It's used to not define the atomsk factory if the binary is not around, see [here](https://github.com/pyiron/pyiron_atomistics/blob/f6007ccb38172a4cd766ff2a7e0614d9cefa6f4a/pyiron_atomistics/atomistics/structure/factory.py#LL35C20-L35C20) and [here](https://github.com/pyiron/pyiron_atomistics/blob/f6007ccb38172a4cd766ff2a7e0614d9cefa6f4a/pyiron_atomistics/atomistics/structure/factory.py#L73).  For sphinx that's probably a bit much, because even when sphinx is not installed as an executable, users might want to analyze calculations run elsewhere.

**Comment by liamhuber** [comment]
Is it not possible to raise these sorts of errors with the import guard Marvin made? I'd also suggest to directly put the "conda install -c conda-forge (package_name)" right in the error, at least for packages where we're the conda feedstock maintainters.

**Comment by samwaseda** [comment]
Yeah right now I'm looking at the best possible place for this error to be raised.

This being said, it sounds also a bit like a general feature that we should have anyway. Currently SPHInX for example raises an error only when it fails to parse the output file, and it also only says that it failed to parse the output file...

**Comment by samwaseda** [comment]
Actually there was a super obvious location: `validate_ready_to_run`.

File: pyiron_atomistics/sphinx/base.py [hunk 1]
```diff
@@ -1827,7 +1836,7 @@ def run_addon(
                 # move output to working directory for successful runs
                 if out.returncode == 0:
                     for file in os.listdir(tempd):
-                        movefile(os.path.join(tempd, file), self.working_directory)
+                        shutil.movefile(os.path.join(tempd, file), self.working_directory)
```
Comment (jan-janssen) [inline pyiron_atomistics/sphinx/base.py]:
```suggestion
                        shutil.move(os.path.join(tempd, file), self.working_directory)
```

Comment (samwaseda) [reply pyiron_atomistics/sphinx/base.py]: ah brilliant

**Pull request closed without merging**